prow/github/client.go

```diff
@@ -2475,6 +2482,34 @@ func (c *client) GetBranches(org, repo string, onlyProtected bool) ([]Branch, er
 	return branches, nil
 }
 
+// GetDefaultBranch returns default branch of the repo.
+// Uses a memory cache for performance, storing repos with key `org_repo_defaultBranch`.
+//
+// See https://docs.github.com/en/rest/reference/repos#list-organization-repositories
+func (c *client) GetDefaultBranch(org, repo string) (string, error) {
+	durationLogger := c.log("GetDefaultBranch", org, repo)
+	defer durationLogger()
+
+	var repoKey = org + "_" + repo + "_defaultBranch"
+	var defaultBranch string
+
+	// Retrieve default branch from cache;
+	// otherwise retrieve repo and set default branch in cache
+	cacheValue, found := memCache.Get(repoKey)
+	if found {
+		defaultBranch = cacheValue.(string)
+	} else {
+		repoObj, err := c.GetRepo(org, repo)
+		if err != nil {
+			return "", err
+		}
+		defaultBranch = repoObj.Repo.DefaultBranch
+		memCache.Set(repoKey, defaultBranch, 0)
```

Comment:
So if you don't expire them this isn't worth a new lib, you can implement a cache as a `map[string]string` with a `sync.Mutex` for thread safety